src/class/cdc/cdc_device.c

```diff
@@ -171,7 +166,7 @@ uint32_t tud_cdc_n_write(uint8_t itf, void const* buffer, uint32_t bufsize)
   uint16_t ret = tu_fifo_write_n(&p_cdc->tx_ff, buffer, bufsize);
 
   // flush if queue more than packet size
-  if ( tu_fifo_count(&p_cdc->tx_ff) >= BULK_PACKET_SIZE )
+  if ( (tu_fifo_count(&p_cdc->tx_ff) >= CFG_TUD_CDC_EP_BUFSIZE) || tu_fifo_full(&p_cdc->tx_ff) )
```

Comment:
shouldn't the packet size check be kept? the buffer may be larger than the packet size, and in that case, we would want to flush when we have enough to fill said packet

Reply:
The buffer size is defined here https://github.com/hathach/tinyusb/blob/master/src/class/cdc/cdc_device.h#L42 , so defining `BULK_PACKET_SIZE` seems as a redefinition...

But maybe I'm just missing reasons for change inhttps://github.com/hathach/tinyusb/pull/547 , I can change it back if you will :)

Reply:
sure, it's redefined, but I'm actually talking about the logic that triggers a flush, you just using the bufsize, but I'm arguing both the bufsize and the packet size should be considered

Reply:
this goes for the zero length packet too, we should see if we sent a full packet and there is no more to send, not if we sent a CFG_TUD_CDC_EP_BUFSIZE